### PR TITLE
Add about tab in settings modal

### PR DESCRIPTION
**Pull request by bbashy**
## Description
Added an about page with some info taken from the README and added links to GitHub and Discord server at the bottom of the tab.

## Related Issue
#654 

## How Has This Been Tested?
I've developed this with a fully running instance.

## Screenshots (if appropriate):
screen shot on 2018-08-08 at 14 40 46

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by noraj** [comment]
I'll try that.

**Comment by wopian** [comment]
Would it better to use the SVG version of the logo instead of an embedded base64 PNG. It adds over 4kb to the client JS while the [SVG is only 1kb unminified](https://github.com/jfurrow/flood/blob/master/client/src/public/safari-pinned-tab.svg), or [500 bytes minified](https://my.mixtape.moe/gqzzzu.svg).

**Comment by bbashy** [comment]
@wopian Deffo. I wasn't sure how to include an svg and where to save it.

File: client/src/javascript/components/modals/settings-modal/AboutTab.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
...fACAaYuYDimy4vSUtFvFE59cBVuvVLyklaDJcUKVokAh5PN7lBgRdIpEoE58fkg8AMiL2N5vGwSr5K1Ilp4j/rioErhtIRqSaE2bvsmtxrPj0oXgg9HMgt2vrFWmz2gEMzEaAOaGZqvs/+tIv3n3H7073R0pUtEORulOlMSUG8sOy4kmQ/lx23+b2mKLva28rOgN1QH5YXYBZGcm/2TyKTkxz+ufeyTi30IBLbU95Nx9LXw55Yi1QAKwyGq4N2Ql+Dt8/ZLWebhAe0bzvQBkxDVwbYlXHspW+je0xjtWMZ4oGyVBTYOgwerAPKAA6WSbuD8hX+p5FIuJ7F7fP9a9SRUB+aH9ApUPC9tlEP6fYBVkspDgVIRtRJQFrAgoAd0iUekTWNXYxRc4nOOeZ8Zko+w/1RgCoKqOR6BEZaJ6Sy/bbRifeWBOiaA4cuPyiyg/GD++SYfYJe2/50tvR3Lyb8/Xf0IBMZCNKrtSuRwJxsxwmgMxUfw7i+qjCG4xO988jRHgArkEyndLRvWK5z88jyKVngJUx1C4cLcgLxE+uV0zULaehHrJ/KxgIe5O/rQMkgFy3HGNeYN1Ht++vONGZFXo1/OCrFwUjLmiE8JOdFyhPTB6eq+a8bs2sR4oh9f7gejZEANmJCe7MaLA8c2K4uDIDruWJhdIzJRB+8jMjTVOzbN/oHKdPHP+m3JSG0toBAL8WUzOVhqnsUX9I+kDdJS5P+EMlH/4OELBYC361yansId27iP/B1SusTZSfOhIOdhHzazc5JZgd99r4tcQyXPyZxPzazo4pn55Tf3+A+hsU1N8hof4WDfX3iKi/SWVA+V0yA+pv01F/n9CA8huVKCi+U4qC4lu1KCi+V6wAtTerB0Ht3fIhUHq7Xg899NBDDz300GNM8R+eMBgKNFW14AAAACV0RVh0ZGF0ZTpjcmVhdGUAMjAxNy0wNS0xNlQyMToyMjowMyswMjowMA6d6FAAAAAldEVYdGRhdGU6bW9kaWZ5ADIwMTctMDUtMTZUMjE6MjI6MDMrMDI6MDB/wFDsAAAAV3pUWHRSYXcgcHJvZmlsZSB0eXBlIGlwdGMAAHic4/IMCHFWKCjKT8vMSeVSAAMjCy5jCxMjE0uTFAMTIESANMNkAyOzVCDL2NTIxMzEHMQHy4BIoEouAOoXEXTyQjWVAAAAAElFTkSuQmCC"/>
```
Comment (jfurrow) [inline client/src/javascript/components/modals/settings-modal/AboutTab.js]: Let's use an SVG as a React component as done for [other icons](https://github.com/jfurrow/flood/tree/master/client/src/javascript/components/icons).

File: client/src/javascript/components/modals/settings-modal/AboutTab.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            <br/>
+            Flood is another web interface for <a href="https://github.com/rakshasa/rtorrent" rel="noopener noreferrer" target="_blank">rtorrent</a>. It implements a Node.js server for communicating with the rTorrent API, storing historical data, and serving the web UI.
```
Comment (jfurrow) [inline client/src/javascript/components/modals/settings-modal/AboutTab.js]: These strings should use `FormattedMessage` like other strings in the app.

[Here's an example of `FormattedMessage`](https://github.com/jfurrow/flood/blob/master/client/src/javascript/components/modals/settings-modal/ConnectivityTab.js#L40-L43), and [here's the corresponding English copy](https://github.com/jfurrow/flood/blob/master/client/src/javascript/i18n/en.js#L140).

File: client/src/javascript/components/modals/settings-modal/AboutTab.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            <br/>
+            <br/>
```
Comment (jfurrow) [inline client/src/javascript/components/modals/settings-modal/AboutTab.js]: Let's not use `<br/>` for spacing — use CSS to apply margin.

File: client/src/javascript/components/modals/settings-modal/SettingsModal.js [hunk 4]
```diff
@@ -220,7 +221,14 @@ class SettingsModal extends React.Component {
...
           scrollContainer: this.modalBodyRef
         }
-      }
+      },
+      about: {
+          content: AboutTab,
+          label: this.props.intl.formatMessage({
+              id: 'settings.tabs.about',
+              defaultMessage: 'About'
+          })
```
Comment (jfurrow) [inline client/src/javascript/components/modals/settings-modal/SettingsModal.js]: There should be an eslint rule for this, but let's use 2 characters for tab width please.

File: client/src/javascript/components/modals/settings-modal/AboutTab.js (hunk 1 above)
Comment (bbashy) [reply client/src/javascript/components/modals/settings-modal/AboutTab.js]: Do we have an SVG of the logo somewhere?

**Comment by bbashy** [comment]
Think someone else should take over this change. I don't know React/JS well enough to do the changes. Sorry.

File: client/src/javascript/components/modals/settings-modal/AboutTab.js (hunk 1 above)
Comment (noraj) [reply client/src/javascript/components/modals/settings-modal/AboutTab.js]: @bbashy Yes there is. It was introduced in https://github.com/jfurrow/flood/pull/363. it is located here: `server/assets/safari-pinned-tab.svg`.

Comment (noraj) [reply client/src/javascript/components/modals/settings-modal/AboutTab.js]: @jfurrow Use Readct baseicon will works only for those icon packs:



Not for the custom flood icon.

Comment (noraj) [reply client/src/javascript/components/modals/settings-modal/AboutTab.js]: I added the SVG logo in PR #694.

You'll be able to use it in a `` or in a container (more responsive).

File: client/src/javascript/components/modals/settings-modal/AboutTab.js (hunk 2 above)
Comment (noraj) [reply client/src/javascript/components/modals/settings-modal/AboutTab.js]: How could it be done with links inside ?

File: client/src/javascript/i18n/en.js [hunk 5]
```diff
@@ -164,6 +164,7 @@ export default {
   'settings.tabs.resources': 'Resources',
   'settings.tabs.authentication': 'Authentication',
   'settings.tabs.userinterface': 'User Interface',
+  'settings.tabs.about': 'About',
```
Comment (noraj) [inline client/src/javascript/i18n/en.js]: I provided translation for French in PR #695.

**Comment by bbashy** [comment]
Could we make this simpler and just import a markdown file? Maybe we can use README.md or make a minified version in about.md

https://github.com/rexxars/react-markdown

**Comment by bbashy** [comment]
I've got it rendering the README.md but the image path won't be found. Any solution to that?

**Comment by noraj** [comment]
@bbashy Don't put the whole README. Just a about.md would be fine. Also the SVG is here https://github.com/jfurrow/flood/pull/694

**Comment by noraj** [comment]
@bbashy I don't know, this is a question for @jfurrow

**Comment by bbashy** [comment]
I'll submit a cleaner PR.

**Pull request closed without merging**